x/farming/keeper/reward.go

```diff
@@ -132,23 +102,42 @@ func (k Keeper) UnmarshalReward(bz []byte) (types.Reward, error) {
 }
 
 // Harvest claims farming rewards from the reward pool account.
-func (k Keeper) Harvest(ctx sdk.Context, farmer sdk.AccAddress, stakingCoinDenoms []string) (sdk.Coins, error) {
+func (k Keeper) Harvest(ctx sdk.Context, farmerAcc sdk.AccAddress, stakingCoinDenoms []string) error {
 	amount := sdk.NewCoins()
 	for _, denom := range stakingCoinDenoms {
-		reward, found := k.GetReward(ctx, denom, farmer)
+		reward, found := k.GetReward(ctx, denom, farmerAcc)
 		if !found {
-			return nil, sdkerrors.Wrapf(types.ErrRewardNotExists, "no reward for staking coin denom %s", denom)
+			return sdkerrors.Wrapf(types.ErrRewardNotExists, "no reward for staking coin denom %s", denom)
 		}
 		amount = amount.Add(reward.RewardCoins...)
 	}
 
-	if err := k.ReleaseStakingCoins(ctx, farmer, amount); err != nil {
-		return nil, err
+	if err := k.ReleaseStakingCoins(ctx, farmerAcc, amount); err != nil {
+		return err
 	}
 
 	for _, denom := range stakingCoinDenoms {
-		k.SetReward(ctx, denom, farmer, sdk.NewCoins())
+		k.DeleteReward(ctx, denom, farmerAcc)
 	}
 
-	return amount, nil
+	if len(k.GetRewardsByFarmer(ctx, farmerAcc)) == 0 {
+		k.GetStakingIDByFarmer(ctx, farmerAcc)
```

Comment:
`k.GetStakingIDByFarmer(ctx, farmerAcc)` is unused with result
and `GetStakingIDByFarmer` can be refactoring to `GetStakingIdByFarmer`